upup/pkg/fi/cloudup/new_cluster.go

```diff
@@ -1173,6 +1173,19 @@ func createEtcdCluster(etcdCluster string, masters []*api.InstanceGroup, encrypt
 		m.InstanceGroup = fi.String(ig.ObjectMeta.Name)
 		etcd.Members = append(etcd.Members, m)
 	}
+
+	// Cilium etcd server is not compacted by the k8s API server.
+	if etcd.Name == "cilium" {
+		if etcd.Manager == nil {
+			etcd.Manager = &api.EtcdManagerSpec{
+				Env: []api.EnvVar{
+					{Name: "ETCD_AUTO_COMPACTION_MODE", Value: "revision"},
+					{Name: "ETCD_AUTO_COMPACTION_RETENTION", Value: "2500"},
```

Comment:
Perhaps we should enable this for existing clusters?

Reply:
The way this is managed with env vars makes it a bit tricky. There is no way of saying "I want to manage this myself".

We could maybe highlight this more in release notes.

Reply:
We could take any value explicitly mentioned in the spec as overriding. That leaves the problem of "I want to omit this environment variable", but that has some possible solutions.

Reply:
Perhaps merge this one and cherry-pick to 1.21 and then have a longer ponder on how to set these by default for existing clusters.

Reply:
The disadvantage of that would be that clusters created in the interim would get these values set in the cluster spec, making it harder to change the defaults later. But perhaps it's an obscure enough feature.

Reply:
These defaults should be fine (same as cilium-etcd-operator used before it got deprecated). For clusters that needs something else, their cluster operators should know what to do.